report_dynamic/models/report_dynamic_section.py

```diff
@@ -0,0 +1,217 @@
+    )
+    model_id_model = fields.Char(
+        string="Model _description", related="report_id.model_id.model"
+    )
+
+    @api.onchange("field_id", "sub_model_object_field_id", "default_value")
+    def onchange_copyvalue(self):
+        self.sub_object_id = False
+        self.copyvalue = False
+        if self.field_id and not self.field_id.relation:
+            self.copyvalue = "${{object.{} or {}}}".format(
+                self.field_id.name, self._get_proper_default_value()
+            )
+            self.sub_model_object_field_id = False
+        if self.field_id and self.field_id.relation:
+            self.sub_object_id = self.env["ir.model"].search(
+                [("model", "=", self.field_id.relation)]
+            )[0]
+        if self.sub_model_object_field_id:
+            self.copyvalue = "${{object.{}.{} or {}}}".format(
+                self.field_id.name,
+                self.sub_model_object_field_id.name,
+                self._get_proper_default_value(),
+            )
+
+    # this then needs to be an onchange, since user
+    # should be able to see preview after setting
+    # a condition, directly.
+    @api.onchange("condition_python")
+    def _compute_condition_python_preview(self):
+        """Compute condition and preview"""
+        for this in self:
+            this.condition_python_preview = False
+            if not (this.resource_ref_model_id and this.res_id and this.resource_ref):
+                continue
+            try:
+                # Check if there are any syntax errors etc
+                this.condition_python_preview = this._eval_condition_python()
+            except Exception as e:
+                # and show debug info
+                this.condition_python_preview = str(e)
+                continue
+
+    def _eval_condition_python(self):
+        if not self.condition_python:
+            return True
+        condition_python = (self.condition_python or "").strip()
+        record = self.resource_ref
+        if not record:
+            return False
+        return safe_eval(condition_python, {"object": record})
+
+    def _eval_condition_domain(self):
+        condition_domain = (self.condition_domain or "[]").strip()
+        return self.resource_ref.filtered_domain(safe_eval(condition_domain))
+
+    def _get_proper_default_value(self):
+        self.ensure_one()
+        is_num = self.field_id.ttype in ("integer", "float")
+        value = 0 if is_num else "''"
+        if self.default_value:
+            if is_num:
+                value = "{}"
+            else:
+                value = "'{}'"
+            value = value.format(self.default_value)
+        return value
+
+    # compute the dynamic content for jinja expression
+    def _compute_dynamic_content(self):
+        # a parent with two children
+        h = self._get_header_object()
+        for this in self:
+            if not (this._eval_condition_python() or this._eval_condition_domain()):
```

Comment:
IMO an `and` is better here, now from start if people add a python condition it doesnt work.. i think we should apply both conditions and dont need to support the "or" case

Reply:
![image](https://user-images.githubusercontent.com/1466356/153230072-5e502546-eec0-4109-9efb-9d88e37ae7c8.png)

(currently this will still render even if name of object = "..." because the domain condition matches)